torchci/components/PeriodicWorkflows.tsx

```diff
@@ -0,0 +1,138 @@
+import { CommitData, JobData } from "lib/types";
+import _ from "lodash";
+import { useState, useEffect } from "react";
+import useSWR from "swr";
+import { fetcherWithToken } from "lib/GeneralUtils";
+import { Octokit } from "octokit";
+
+const SUPPORTED_WORKFLOWS: { [k: string]: any } = {
+  "pytorch/pytorch": {
+    trunk: "Run trunk jobs",
+    periodic: "Run periodic jobs",
+    slow: "Run slow jobs",
+  },
+};
+
+function hasWorkflow(jobs: JobData[], workflow: string) {
+  return _.find(
+    jobs,
+    (job) => job.name !== undefined && job.name.startsWith(workflow)
+  );
+}
+
+function PeriodicWorkflow({
```

Comment:
Should it be a workflow dispatch, to match github terminology?

Reply:
Interesting, let me see if I can redo the change to use workflow dispatch.  It would be better to be able to achieve this functionality without creating new `cilfow` tags if it's not absolutely needed

Reply:
Rename the new API to `/api/github/dispatch/[repoOwner]/[repoName]/[workflow]/[sha]`.  On another note, it looks like I would need to setup something to clean up these refs as there is no closing PR event to hook into (https://github.com/pytorch/test-infra/blob/main/torchci/lib/bot/ciflowPushTrigger.ts#L118-L125).  A cron job like stale.yml on PyTorch would probably work.